onnx/defs/shape_inference.h

```diff
@@ -312,12 +313,24 @@ inline void propagateShapeFromInputToOutput(
       ctx.getInputType(inputIndex)->tensor_type().shape();
 }
 
-inline void propagateShapeAndTypeFromFirstInput(InferenceContext& ctx) {
-  propagateElemTypeFromInputToOutput(ctx, 0, 0);
-  if (!hasNInputShapes(ctx, 1)) {
+inline void propagateShapeAndTypeFromInputToOutput(InferenceContext& ctx, size_t inputIndex, size_t outputIndex) {
+  propagateElemTypeFromInputToOutput(ctx, inputIndex, outputIndex);
```

Comment:
Check whether we have the type information for input[inputIndex] first.

Reply:
According to https://github.com/onnx/onnx/blob/765f5ee823a67a866f4bd28a9860e81f3c811ce8/onnx/onnx.proto#L425 there should be. If there isn't the function will throw which I deem OK

Reply:
https://github.com/onnx/onnx/blob/765f5ee823a67a866f4bd28a9860e81f3c811ce8/onnx/onnx.proto#L143, we may not have valueinfo for some intermediate value. In that case, the type information is not available.

Reply:
And this shall be changed from a hard failure (https://github.com/onnx/onnx/blob/84a0441ae28795a928005863dc142bee81827566/onnx/defs/shape_inference.h#L212) to a soft failure (do nothing)?   
Just asking again as this changes existing behaviour.

Reply:
if you cannot do the shape inference, it's not a good idea to throw exception, instead you should just skip it. This logic is just to avoid segfault.